File: app/src/main/java/domon/cn/gankio/contract/GirlsContract.java
package domon.cn.gankio.contract;

import domon.cn.gankio.BasePresenter;
import domon.cn.gankio.BaseView;
import domon.cn.gankio.data.GankGirlsData;

/**
 * Created by Domon on 16-10-28.
 */

public interface GirlsContract {

    interface View extends BaseView<Presenter> {
        void getGankGirlsData();

        void setData(GankGirlsData gankGirlsData);
    }

    interface Presenter extends BasePresenter {
        void reqGrilsGankData(String index, String count);
    }
}

File: app/src/main/java/domon/cn/gankio/contract/GirlsContract.kt
package domon.cn.gankio.contract

import domon.cn.gankio.BasePresenter
import domon.cn.gankio.BaseView
import domon.cn.gankio.data.GankGirlsData

/**
 * Created by Domon on 16-10-28.
 */

interface GirlsContract {

    interface View : BaseView<Presenter> {
        fun getGankGirlsData()

        fun setData(gankGirlsData: GankGirlsData)
    }

    interface Presenter : BasePresenter {
        fun reqGrilsGankData(index: String, count: String)
    }
}
